### PR TITLE
Feature/ar 578 create or edit screen 07 2021

### PR DESCRIPTION
**Pull request by sinejespersen**
https://jira.itkdev.dk/browse/AR-578

image

image

image

What i did do

* added react-beautiful-dnd for drag and drop component
* added react-multi-select-component for multiselect component
* added additional fixture data
* added todo-css 
* added validations for forms on edit-tag, edit-screen(new), edit-category
* added formcomponents: radio buttons, select & multiselect
* multiselect has: location dropdown, groups dropdown and playlists dropdown
* added tests
* added drag and drop table, for sorting the playlists

What i didn't do
* add slides-dropdown in the bottom of edit-screen, as i simply did not understand the wireframes and need clearance
* add modal for quickedit ... this will be added in a future pr!

### CODE REVIEW

File: admin/src/components/util/forms/multiselect-dropdown/tags/tag-dropdown.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import MultiSelectComponent from "../multi-dropdown";
+/**
```
Comment (cableman) [inline admin/src/components/util/forms/multiselect-dropdown/tags/tag-dropdown.jsx:4]: Missing empty line

File: admin/src/components/util/forms/multiselect-dropdown/tags/tag-dropdown.jsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+          isCreatable
+          name={name}
+          errors={errors}
+        />
+      )}
+    </>
+  );
+}
+
+TagDropdown.propTypes = {
```
Comment (cableman) [inline admin/src/components/util/forms/multiselect-dropdown/tags/tag-dropdown.jsx:56]: `TagDropdown.propTypes` also defined in line 71 ?

File: admin/src/components/util/helpers/content-string.spec.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import contentString from "./content-string";
```
Comment (cableman) [inline admin/src/components/util/helpers/content-string.spec.js:1]: Unit test ? Is this cypress test?

File: admin/src/components/util/helpers/form-errors-helper.spec.js [hunk 4]
```diff
@@ -0,0 +1,122 @@
+describe("Form validations", () => {
+  it("It validates new tag", () => {
+    cy.visit("/tag/new");
```
Comment (cableman) [inline admin/src/components/util/helpers/form-errors-helper.spec.js:3]: Test of form error helper requires that `/tag/new` exits (hidden dependency) can we do testing without this dependency?

File: docker-compose.yml [hunk 5]
```diff
@@ -14,7 +14,7 @@ services:
     networks:
       - app
     ports:
-      - '3306'
+      - "3306"
```
Comment (cableman) [inline docker-compose.yml:17]: We should not change this to "" as it don't match when updating later on with templates from itkdev docker setup

File: docker-compose.yml [hunk 6]
```diff
@@ -86,11 +86,11 @@ services:
     depends_on:
       - nginx
     environment:
-      - CYPRESS_VIDEO=false
+      - CYPRESS_VIDEO=true
```
Comment (cableman) [inline docker-compose.yml:89]: Don't thing video should be enabled as default

**Review by cableman — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 5 above)
Comment (sinejespersen) [reply docker-compose.yml:17]: sorry, my linter

**Review by cableman — APPROVED** [review]

**Pull request merged**